ffi/derive/src/wrapper.rs

```diff
@@ -0,0 +1,119 @@
+use proc_macro2::TokenStream;
+use proc_macro_error::abort;
+use quote::quote;
+
+use crate::{
+    ffi_fn,
+    impl_visitor::{Arg, FnDescriptor, ReturnArg},
+    util::{gen_arg_ffi_to_src, gen_arg_src_to_ffi},
+};
+
+pub fn wrap_as_opaque(input: &syn::DeriveInput) -> TokenStream {
+    let attrs = &input.attrs;
+    let vis = &input.vis;
+    let ident = &input.ident;
+
+    let item_type = match input.data {
+        syn::Data::Enum(_) => quote! {enum},
+        syn::Data::Struct(_) => quote! {struct},
+        syn::Data::Union(_) => quote! {union},
+    };
+
+    match &input.data {
+        syn::Data::Enum(_) | syn::Data::Struct(_) => {
+            quote! {
+                #(#attrs)*
+                #vis #item_type #ident{
+                    ptr: [u8; 0],
```

Comment:
Why do we have type `[u8; 0]` for pointer?

Reply:
I believe I fixed it now